benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -544,15 +535,25 @@ void GraphicsBenchmarkApp::SetupSkyBoxMeshes()
 
 void GraphicsBenchmarkApp::SetupSphereMeshes()
 {
-    OrderedGrid grid(kMaxSphereInstanceCount, kSeed);
+    const uint32_t requiredSphereCount = std::max<uint32_t>(pSphereInstanceCount->GetValue(), kDefaultSphereInstanceCount);
+    const uint32_t initSphereCount     = std::min<uint32_t>(kMaxSphereInstanceCount, 2 * std::max(requiredSphereCount, mInitializedSpheres));
+
+    OrderedGrid grid(initSphereCount, kSeed);
+    mInitializedSpheres = initSphereCount;
+
+    // Destroy the meshes if they were created.
+    for (auto& mesh : mSphereMeshes) {
+        if (mesh != nullptr) {
+            GetDevice()->DestroyMesh(mesh);
```

Comment:
nit: reset it back to nullptr after you destroyed it, try not to leave dangling pointer around.

Reply:
Done.